storage/store.go

```diff
@@ -1007,8 +1045,19 @@ func (s *Store) GroupStorage(groupID uint64) multiraft.WriteableGroupStorage {
 	defer s.mu.RUnlock()
 	r, ok := s.ranges[int64(groupID)]
 	if !ok {
-		log.Warningf("%p requested nonexistent range with raft ID %d", s, groupID)
-		return nil
+		var err error
+		r, err = NewRange(&proto.RangeDescriptor{
+			RaftID: int64(groupID),
```

Comment:
Yeah, I suppose we should do that refactoring. But wow, I dislike unsigned ints. I really just don't see the point. Could someone explain to me why people bother? Just because they can't be negative has always struck me as about as helpful as "const" in C++. Yes...there's utility, but you've just introduced another thing, sprinkled here and there.

There are overflow and casting issues between the two, but that has nothing to do with their broad usage. I feel like people see them as "safer" in some meaningful way which I just don't see. Is there something I'm missing?